content/ai/concepts/chatgpt/chatgpt.md

```diff
@@ -0,0 +1,103 @@
+
+### What is the relation to Internet Search?
+
+`ChatGPT` is trained by using vast quantities of data from the Internet. However LLMs are different from Internet search engines:
+
+- No ongoing daily web crawlers and updates
+- Data can be out of date. `ChatGPT 3` uses data from 2021.
+- The size of the data trained in the model
+- ChatGPT uses a type of fine-tuning called Reinforcement Learning from Human Feedback (RLHF)
+- ChatGPT-specific approaches including `prompts`, `embeddings`, and parameters such as `temperature`.
+
+## History of ChatGPT
+
+There were several generations of AI technologies that have led up to ChatGPT.
+
+### Feed Forward Networks
+
+Artificial Neural Networks (ANNs) are a class of computational functions inspired by how neurons work in the human brain. Neurons are arranged in layers and training occurs by guiding the network to know correct from incorrect answers. These are called feed-forward networks because signals move forward through the system.
+
+One example of a common ANN is called Convolutional Neural network (CNN). These are commonly used in image recognition. You can think of a convolution as a filter applied to the image to make things like edges or specific features stand out.
+
+### Recurrent Neural Networks
+
+One limitation of ANNs is that they don’t have the concept of time sequences and the input size is fixed. So for language processing where you have a sequence of words or a sentence, Recurrent Neural Networks (RNNs) were created.
+
+RNNs feed decision results back in the system. So for example if we wanted to translate the sentence “What time is it?” by the time the word “it” is encountered, the system would already know all about the question.
+
+While RNNs have been very useful in doing language translation, they do not scale well with large amounts of information. Also computation can not be done in parallel.
+
+### Breakthrough with Transformers
+
+The breakthrough that allowed BERT and ChatGPT-based models is called `Transformers`. This was introduced in a paper from 2019 entitled `“Attention is all you need”`. It was presented as a way to build machine translation but to have much faster parallel execution. It also has a new concept called `Self Attention` to further make sense of long language sequences. It creates connections between different parts of the sequence.
+
+So how does it all really work? In the `Transformer` based model each word is transformed into a token. As tokens are read as part of an input sentence the model starts predicting what the output should be - similar in concept to sentence completion.
+
+Due to the use of `Transformers` in creating the model, the potential output is not just a prediction of the next word. Because the model is trained with deep understanding and `Attention` whole sentences and groupings of text are generated. The model used by ChatGPT can accept several thousand tokens as input .
```

Comment:
```suggestion
Due to the use of transformers in creating the model, the potential output is not just a prediction of the next word. Because the model is trained with deep understanding and "attention", whole sentences and groupings of text are generated. The model used by ChatGPT can accept several thousand tokens as input.
```

Reply:
Resolved issues